File: CPP/Lesson_105_merge_strings_alternately.cpp
// =============================================================
// MIT License | @analyticswithharry2026
// GitHub  : https://github.com/analyticswithharry
// YouTube : Analytics with Harry
// =============================================================
// Lesson     : 105 -- Merge Strings Alternately
// Category   : Two Pointers
// Difficulty : Easy
// Study Plan : Day 53
// =============================================================
//
// QUESTION:
//   Given two strings, merge them by adding letters in alternating order, starting with word1. If one is longer, append the rest.
// =============================================================

#include <vector>
#include <string>
#include <iostream>
#include <stack>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <map>
#include <set>
#include <algorithm>
#include <climits>
#include <numeric>
#include <functional>
#include <cmath>
using namespace std;
class Solution { public:
    string mergeAlternately(string a, string b){
        string r; int i=0;
        while (i<(int)a.size() || i<(int)b.size()){
            if (i<(int)a.size()) r+=a[i];
            if (i<(int)b.size()) r+=b[i];
            i++;
        }
        return r;
    }
};
int main(){ Solution s; cout<<s.mergeAlternately("abc","pqr")<<" "<<s.mergeAlternately("ab","pqrs")<<endl; }

File: Java/Lesson105_MergeStringsAlternately.java
// =============================================================
// MIT License | @analyticswithharry2026
// GitHub  : https://github.com/analyticswithharry
// YouTube : Analytics with Harry
// =============================================================
// Lesson     : 105 -- Merge Strings Alternately
// Category   : Two Pointers
// Difficulty : Easy
// Study Plan : Day 53
// =============================================================
//
// QUESTION:
//   Given two strings, merge them by adding letters in alternating order, starting with word1. If one is longer, append the rest.
// =============================================================

public class Lesson105_MergeStringsAlternately {
    public String mergeAlternately(String a, String b){
        StringBuilder sb = new StringBuilder(); int i=0;
        while (i<a.length() || i<b.length()){
            if (i<a.length()) sb.append(a.charAt(i));
            if (i<b.length()) sb.append(b.charAt(i));
            i++;
        }
        return sb.toString();
    }
    public static void main(String[] a){
        Lesson105_MergeStringsAlternately x = new Lesson105_MergeStringsAlternately();
        System.out.println(x.mergeAlternately("abc","pqr"));
        System.out.println(x.mergeAlternately("ab","pqrs"));
    }
}
